File: mypostbank-ui/src/app/services/models/transaction-dto.ts
/* tslint:disable */
/* eslint-disable */
export interface TransactionDto {
  accountId?: number;
  amount?: number;
  beneficiary?: number;
  cardId?: number;
  date?: string;
  description?: string;
  id?: number;
  transactionType?: 'TRANSFER' | 'CARD_PAYMENT' | 'WITHDRAWAL' | 'DEPOSIT' | 'CREDIT_INSTALLMENT';
}

File: src/main/java/com/Spring/MyPostBank/DTOs/TransactionDTO.java
package com.Spring.MyPostBank.DTOs;

import com.Spring.MyPostBank.Enums.TransactionType;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TransactionDTO {
    private Integer id;
    private TransactionType transactionType;
    private BigDecimal amount;
    private String description;
    private LocalDateTime date;
    //receiver
    private Integer beneficiary;
    private Integer cardId;
    //sender
    private Integer accountId;
}
